### PR TITLE
Add spin buttons

### LINKED ISSUE #3928: Implement button:loading style
<a href="https://github.com/JohnONolan"></img></a> **Issue by [JohnONolan](https://github.com/JohnONolan)**
_Thursday Jul 17, 2014 at 09:29 GMT_
_Originally opened as https://github.com/TryGhost/Ghost-UI/issues/71_

----

Let's get rid of the - pardon me - fucking horrendous button style that exists right now when an action is pending. (Where the button goes dim + inactive, but still has a hover state, but doesn't do anything)

Can we bring over the button-loading spinner from ghost.org and use that instead?

### PR DESCRIPTION
**Pull request by acburdine**
closes #3928 

credit goes to @halfdan for the initial implementation (couldn't cherry pick the commit correctly - sorry!)

### CODE REVIEW

File: core/client/app/templates/components/gh-editor-save-button.hbs [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-</button>
+{{gh-spin-button type="button" class=buttonClass action="save" buttonText=saveText}}
 
 {{#gh-dropdown-button dropdownName="post-save-menu" classNameBindings=":btn :btn-sm isDangerous:btn-red:btn-blue btnopen:active :dropdown-toggle :up"}}
     <i class="options icon-arrow2"></i>
     <span class="sr-only">Toggle Settings Menu</span>
 {{/gh-dropdown-button}}
 {{#gh-dropdown name="post-save-menu" closeOnClick="true" tagName="div" classNames="dropdown editor-options"}}
-    <ul class="dropdown-menu dropdown-triangle-bottom-right">
+    <ul class="dropdown-menu pull-right dropdown-triangle-bottom-right">
```
Comment (acburdine) [inline core/client/app/templates/components/gh-editor-save-button.hbs]: @JohnONolan I'm not sure if this was intentional or not (mobile perhaps?), but this dropdown looked off, so I added the pull-right class.

**Comment by halfdan** [comment]
@acburdine Yea there was no way you could've cherry-picked that ;) Don't worry. Glad you picked it up!

**Comment by acburdine** [comment]
@ErisDS @kevinansfield @jaswilli I think I got all of the necessary buttons added

One thing I didn't do was add spin-buttons to the modals that had ajax calls. I initially was going to, but it turned out to be super complex, and IMO overkill, especially because the modals close when the promise resolves. However, I can add them in if it is necessary

**Comment by ErisDS** [comment]
I just checked this out and it's looking neat :+1: 

Is there a way to not stop the spinner before transitioning? I noticed on sign-in that the spinner stops, and then the page transitions. I think it would feel nicer if it was still spinning when the transition happened?

**Comment by acburdine** [comment]
I think it's possible. I'll try some things with Ember Simple Auth and see if I can make it work. I also need to figure out what's causing these tests to fail.

**Pull request merged**